addons/html_builder/static/src/builder/builder_sidebar/invisible_elements_panel.js

```diff
@@ -72,9 +72,28 @@ export class InvisibleElementsPanel extends Component {
     }
 
     toggleElementVisibility(invisibleEntry) {
-        const snippetEl = invisibleEntry.snippetEl;
-        const isVisibleEl = isSnippetVisible(snippetEl);
-        setSnippetVisibility(invisibleEntry, !isVisibleEl);
+        const toggleVisibility = (snippetEl) => {
+            let isVisibleEl = isSnippetVisible(snippetEl);
+            setSnippetVisibility(invisibleEntry, !isVisibleEl);
+            isVisibleEl = !isVisibleEl;
+            const dispatchName = isVisibleEl ? "on_target_show" : "on_target_hide";
+            this.env.editor.resources[dispatchName].forEach((cb) => cb(snippetEl));
+            this.env.editor.resources["update_containers"].forEach((cb) => cb(snippetEl));
+            // TODO _disableUndroppableSnippets
+        };
+
+        // Toggle all its descendants to invisible (Hide)
+        if (invisibleEntry.isVisible) {
+            invisibleEntry.children.forEach((child) => {
+                if (child.isVisible) {
+                    this.toggleElementVisibility(child);
+                }
+            });
+        } else if (invisibleEntry.parents && !invisibleEntry.parents.isVisible) {
+            // Toggle all its parents to visible (show)
+            this.toggleElementVisibility(invisibleEntry.parents);
+        }
```

Comment:
Maybe add more explanation in the comments so we directly understand what it does, like `When toggling the visibility of an element to "Hide", also toggle all its descendants` and  `When toggling the visibility of an element to "Show", also toggle all its parents` :slightly_smiling_face: